singer_sdk/helpers/_flattening.py

```diff
@@ -430,7 +430,7 @@ def _flatten_record(
             items.append(
                 (
                     new_key,
-                    json.dumps(v, use_decimal=True, default=str)
+                    json.dumps(v, use_decimal=True, default=str, allow_nan=True)
```

Comment:
Wdyt about making this configurable with two initial NaN options (e.g. `fail` and `allow`)?

Reply:
I gave it a first try, please see move the conversation to #2222 and close this PR if possible.

Reply:
@edgarrmondragon  can you take a look?

Reply:
@edgarrmondragon  please close this PR and see the proposal in https://github.com/meltano/sdk/pull/2222